### ISSUE TITLE
Firefox 57+ support

### ISSUE THREAD

**Issue by bvanderhaar** [NickSC/nzbFox#30]
On November 14 Firefox is moving to WebExtensions for Add Ons.  Is there work being done to get this extension migrated?  What can we do to help?

**Comment by flxfxp** [comment]
Ditto, lmk what needs to be done, how reliant is nzbFox on XUL?

**Comment by NickSC** [comment]
I haven't had time to look in to how much work is required to migrate it to WebExtensions. I presume the thing that will break is mostly the https://github.com/NickSC/nzbFox/blob/master/lib/chrome-mod.js file which is used to insert the "Send to ___" option in nzb download popup windows. I don't know if this functionality is available via WebExtensions since it was pretty hacky to begin with.

**Comment by bvanderhaar** [comment]
I'm OK with cutting the Sent to and toolbar icon to start out; my main use case is using the NZB sites to just quickly add the NZB to NZBget.
According to NZBget documentation, the append API method (https://nzbget.net/api/append) is mainly used to add an NZB to nzbget with the NZB URL - this should be fairly straightforward (and is probably in this file - https://github.com/NickSC/nzbFox/blob/master/lib/NZBGet_API.js)
My next investigation is finding out how to modify the DOM to add the nzbFox icon & if the method used here is still supported in the new WebExtension.
I have very limited time; but I'm willing to pitch in some time to help out and/or test.  I rely on this plugin fairly heavily in Firefox now.

**Comment by bylokk** [comment]
yes, porting this addon with reduced functionality is better than having no add-on anymore. I'm looking forward to seeing this add-on again in the future.

**Comment by ChrisVanBael** [comment]
I would also love to have this addon available on 57

**Comment by Minedune** [comment]
Is there a way to donate? We should all contribute to get ball rolling such a great add on
